src/GitStatusCache/src/Git.cpp

```diff
@@ -270,6 +271,20 @@ bool Git::GetRefStatus(Git::Status& status, UniqueGitRepository& repository)
 		Log("Git.GetRefStatus.NoUpstream", Severity::Spam)
 			<< R"(Branch does not have a remote tracking reference. { "repositoryPath": ")" << status.RepositoryPath
 			<< R"(", "localBranch": ")" << status.Branch << R"(" })";
+
+		auto upstreamName = git_buf{ 0 };
+		result = git_branch_upstream_remote(

```

Comment:
Seems reasonable. I'll tweak to check both. At that point I think we'll have all the information to populate the Upstream even if it's gone, so I'll adjust the log statement and I think I can revert the inversion on the posh-git change.